pymeasure/instruments/hp/hp34401A.py

```diff
@@ -49,3 +49,229 @@ def __init__(self, adapter, **kwargs):
             asrl={'baud_rate': 9600, 'data_bits': 7, 'parity': 2},
             **kwargs
         )
+
+    function_ = Instrument.control(
+        "FUNC?", "FUNC \"%s\"",
+        """ A string property that controls the measurement function,
+        which can take the values: DCV, DCV_RATIO, ACV, DCI, ACI,
+        R2W, R4W, FREQ, PERIOD, CONTINUITY, DIODE. """,
+        validator=strict_discrete_set,
+        values=FUNCTIONS,
+        map_values=True,
+        get_process=lambda v: v.strip('"'))
+
+    @property
```

Comment:
I think you should be able to make these properties as pymeasure properties.
Maybe you could use the `command_process` parameter to inject the current range prefix into the Command.

Similarly to the properties below.

Reply:
The problem with `command_process` is that it has to be completely static which the currently active function is not. I think I had this problem before...
The only solution I know of for this problem is to use custom properties. Do you know of another one?

Of course I could also add pymeasure properties for each combination of function and (range, autorange, resolution, nplc) like in the 34450A driver, but I think this is nonsense. This would add 44 properties with a lot of duplication and you most certainly will always use the one for the currently active function.

Reply:
> This would add 44 properties with a lot of duplication and you most certainly will always use the one for the currently active function.

No, don't make it that complicated.

I see the point. The properties are not dynamic.

You could use something similar to the channel ids: you add curly braces in the command string and replace them in the write method.
Just an idea.

If we don't find a better solution than python properties, they are the best 😁

Reply:
Wow, why didn't I think of that... I now added the implementation using a customized `write()` method and this actually makes the code much cleaner and 20 lines shorter. This is in fact exactly what I was searching for. So thanks for this suggestion, I will keep it in mind in the future :)